nntrainer/layers/zoneout_lstmcell.h

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 hyeonseok lee <hs89.lee@samsung.com>
+ *
+ * @file   zoneout_lstmcell.h
+ * @date   30 November 2021
+ * @brief  This is ZoneoutLSTMCell Layer Class of Neural Network
+ * @see	   https://github.com/nnstreamer/nntrainer
+ * @author hyeonseok lee <hs89.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __ZONEOUTLSTMCELL_H__
+#define __ZONEOUTLSTMCELL_H__
+#ifdef __cplusplus
+
+#include <acti_func.h>
+#include <common_properties.h>
+#include <layer_context.h>
+#include <layer_impl.h>
+#include <lstmcell_core.h>
+
+namespace nntrainer {
+
+/**
+ * @class   ZoneoutLSTMCellLayer
+ * @brief   ZoneoutLSTMCellLayer
+ */
+class ZoneoutLSTMCellLayer : public LayerImpl {
+public:
+  /**
+   * @brief HiddenStateZoneOutRate property, this defines zone out specification
+   * of layer
+   *
+   */
+  class HiddenStateZoneOutRate : public nntrainer::Property<float> {
```

Comment:
The definition can move to the zoneout_lstmcell.cpp, only declaration here will be enough, for a later PR.